make navigation header help dropdown ui consistent

paulinashakirova opened the pull request:
## Summary
This PR fixes [Make Help dropdown consistent across all environments](https://github.com/elastic/kibana/issues/199465) issue.

Since we need to make dropdown consistent also in the cloud, I will firstly merge this PR, and then deal with the Cloud part by either opening PR in that repo, or reach out to the responsible team showcasing changes in Kibana and ask them to implement the changes.

This is how the dropdown is displayed currently (1st screenshot) and how it will be looking with my changes (2nd screenshot).

After discussing the design with @ek-so and trying out different variants, this seems to be the most suitable and universal.

The changes include refactoring usage or relevant Eui components and removing the displaying of icons in the general menu dropdown (while keeping this functionality of adding icons to the solutions additional menu dropdown items).

<img width="330" alt="Screenshot 2025-01-07 at 13 07 49" src="https://github.com/user-attachments/assets/a01a4966-dc6d-4c7c-86c2-4588960237f1" />

<img width="309" alt="Screenshot 2025-01-07 at 13 11 06" src="https://github.com/user-attachments/assets/18271f34-d374-473a-bf89-a63d127c7a81" />

Commit:
make navigation header help dropdown ui constistent

Commit:
make navigation header help dropdown ui constistent

paulinashakirova commented on packages/core/chrome/core-chrome-browser-internal/src/ui/header/header_help_menu.tsx:
```diff
@@ -196,7 +194,7 @@ export class HeaderHelpMenu extends Component<Props, State> {
           {customContent && (
             <>
-              <EuiHorizontalRule margin="m" />
+              <EuiPopoverFooter css={{ padding: '8px' }} />
```
After trying different options - using the Footer for the additional, optional bottom menu seems to be the only option to have the separation with horizontal lines as per requested design.
With the current implementation we had to use a custom horizontal rule, whereas PopoverFooter has this divider line out of the box.
The custom padding is an equivalent of the margin="m".

paulinashakirova commented on packages/core/chrome/core-chrome-browser-internal/src/ui/header/header_help_menu.tsx:
```diff
@@ -210,40 +208,37 @@ export class HeaderHelpMenu extends Component<Props, State> {
-                </EuiButtonEmpty>
-                {!isLast && <EuiSpacer size="xs" />}
-              </Fragment>
+        {defaultContentLinks.map(({ href, title, onClick: _onClick, dataTestSubj }, i) => {
```
Removed iconType here.

Commit:
new main, but without my changes for this pr

Commit:
new main, but without my changes for this pr

Commit:
correct location with my changes

Commit:
correct location with my changes

paulinashakirova commented on src/core/packages/chrome/browser-internal/src/ui/header/header_help_menu.tsx:
```diff
@@ -196,7 +194,7 @@ export class HeaderHelpMenu extends Component<Props, State> {
           {customContent && (
             <>
-              <EuiHorizontalRule margin="m" />
+              <EuiPopoverFooter css={{ padding: '8px' }} />
```
After trying different options - using the Footer for the additional, optional bottom menu seems to be the only option to have the separation with horizontal lines as per requested design.
With the current implementation we had to use a custom horizontal rule, whereas PopoverFooter has this divider line out of the box.
The custom padding is an equivalent of the margin="m".

paulinashakirova commented on src/core/packages/chrome/browser-internal/src/ui/header/header_help_menu.tsx:
```diff
@@ -210,40 +208,37 @@ export class HeaderHelpMenu extends Component<Props, State> {
-                </EuiButtonEmpty>
-                {!isLast && <EuiSpacer size="xs" />}
-              </Fragment>
+        {defaultContentLinks.map(({ href, title, onClick: _onClick, dataTestSubj }, i) => {
```
removed iconType here

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/264884)
* Commit: 8252651a2421bbf421b631a79776c753cdeeaf7b

### Failed CI Steps
* [Jest Tests #21](https://buildkite.com/elastic/kibana-pull-request/builds/264884#019441b2-ee54-4f46-950c-059b7fcbf160)
* [Jest Tests #21](https://buildkite.com/elastic/kibana-pull-request/builds/264884#019441cd-1383-4a5d-a42e-107763cb4f43)

cc @paulinashakirova

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264884","number":264884,"commit":"8252651a2421bbf421b631a79776c753cdeeaf7b"}],"number":264884}
buildkite-pr-comment-->

Dosant commented on src/core/packages/chrome/browser-internal/src/ui/header/header_help_menu.tsx:
```diff
@@ -196,7 +194,7 @@ export class HeaderHelpMenu extends Component<Props, State> {
           {customContent && (
             <>
-              <EuiHorizontalRule margin="m" />
+              <EuiPopoverFooter css={{ padding: '8px' }} />
```
if possible, let's use a const from EUI theme here?

paulinashakirova replied:
Great suggestions! ✅

Dosant reviewed (commented):
~Seems like serverless is using some other menu? Probably that one also should be updated? If it is not too problematic, then would be even nicer if they'd use the same component~ My bad, was looking at different instance. Serverless also looks code to me. I'd ask for @ek-so review from design perspective

Commit:
useEuithemeVar for padding value

Commit:
useEuithemeVar for padding value

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/265197)
* Commit: 964c1ca0fdb54c13ab25a6c80c1ad86ea774fbfb

### Failed CI Steps
* [Jest Tests #19](https://buildkite.com/elastic/kibana-pull-request/builds/265197#0194467d-609c-46a3-a1cb-3d45a37b8c0a)
* [FTR Configs #111](https://buildkite.com/elastic/kibana-pull-request/builds/265197#0194467d-60f1-456e-b1b0-19005a031db3)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/265197#0194467d-609c-46a3-a1cb-3d45a37b8c0a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/265197/jobs/0194467d-609c-46a3-a1cb-3d45a37b8c0a/artifacts/019446ae-43fb-4a63-ac9b-546040848d4a) Jest Tests #<span></span>19 / StepDefinePackagePolicy default API response should display vars coming from package policy

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/3b2ab7b4134c05c00a74a70d7905f6b06e1364c1) | [after](https://github.com/elastic/kibana/commit/964c1ca0fdb54c13ab25a6c80c1ad86ea774fbfb) | diff |
| --- | --- | --- | --- |
| `@kbn/core-chrome-browser` | 104 | 103 | -1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/3b2ab7b4134c05c00a74a70d7905f6b06e1364c1) | [after](https://github.com/elastic/kibana/commit/964c1ca0fdb54c13ab25a6c80c1ad86ea774fbfb) | diff |
| --- | --- | --- | --- |
| `core` | 444.5KB | 444.5KB | -35.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/3b2ab7b4134c05c00a74a70d7905f6b06e1364c1) | [after](https://github.com/elastic/kibana/commit/964c1ca0fdb54c13ab25a6c80c1ad86ea774fbfb) | diff |
| --- | --- | --- | --- |
| `@kbn/core-chrome-browser` | 211 | 210 | -1 |


</details>


### History
* :broken_heart: [Build #265096](https://buildkite.com/elastic/kibana-pull-request/builds/265096) failed de26711c23519e22f42ad687e5d66ffd48f92b0d
* :broken_heart: [Build #264884](https://buildkite.com/elastic/kibana-pull-request/builds/264884) failed 8252651a2421bbf421b631a79776c753cdeeaf7b

cc @paulinashakirova

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265197","number":265197,"commit":"964c1ca0fdb54c13ab25a6c80c1ad86ea774fbfb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265096","number":265096,"commit":"de26711c23519e22f42ad687e5d66ffd48f92b0d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264884","number":264884,"commit":"8252651a2421bbf421b631a79776c753cdeeaf7b"}],"number":265197}
buildkite-pr-comment-->

Dosant commented on src/core/packages/chrome/browser-internal/src/ui/header/header_help_menu.tsx:
```diff
@@ -196,7 +195,7 @@ export class HeaderHelpMenu extends Component<Props, State> {
           {customContent && (
             <>
-              <EuiHorizontalRule margin="m" />
+              <EuiPopoverFooter css={{ padding: euiThemeVars.euiSizeS }} />
```
@tsullivan, could you please double check if this is ok to use variable like this? Normally, I'd use variables from context, but here we have a class component and accessing context can be a bit annoying

Dosant replied:
@paulinashakirova based on this [issue and insturctions](https://github.com/elastic/kibana/issues/199715#json-tokens), I think using euiThemeVars from that package is deprecated and we instead should use the from context. 

To use the theme from context we'd normally use `useEuiTheme()` hook, but since this is a class component need to access context with hoc: https://eui.elastic.co/#/theming/theme-provider#consuming-with-the-react-hoc

paulinashakirova replied:
Great, changed, thank you!

Dosant reviewed (approved):
Code lgtm. I'll defer to @ek-so for design review 🙏

ek-so commented:
> Code lgtm. I'll defer to @ek-so for design review 🙏

Hey, I took a look already. Asked Paulina one small thing, waiting for reply. 
I'll post here for visibility:
Are spacers between those items "baked" into euiPopover? Which means, we can't change them (make them "none")?
I wasn't able to find such a prop in eui documents, but just to confirm.
![CleanShot 2025-01-09 at 11 51 07@2x](https://github.com/user-attachments/assets/4133d35f-63e0-4ab5-a6ff-ed2c58a08c2e)

Commit:
remove spacer per design

Commit:
remove spacer per design

ek-so reviewed (approved):
LGTM

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/265649)
* Commit: 6a71bd959d864b132bf05f4ed33d921dc8163462

### Failed CI Steps
* [FTR Configs #67](https://buildkite.com/elastic/kibana-pull-request/builds/265649#01944b97-64bf-4f7c-8efc-645894d35fcc)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/265649#01944b97-64bf-4f7c-8efc-645894d35fcc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/265649/jobs/01944b97-64bf-4f7c-8efc-645894d35fcc/artifacts/01944bb5-b74a-41cb-a400-51f40b16cac8) FTR Configs #<span></span>67 / Cloud Security Posture Test adding Cloud Security Posture Integrations CNVM CNVM AWS Hyperlink on PostInstallation Modal should have the correct URL

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/45f3241db0bd06a264a94d8cfd104bf4057db77e) | [after](https://github.com/elastic/kibana/commit/6a71bd959d864b132bf05f4ed33d921dc8163462) | diff |
| --- | --- | --- | --- |
| `@kbn/core-chrome-browser` | 104 | 103 | -1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/45f3241db0bd06a264a94d8cfd104bf4057db77e) | [after](https://github.com/elastic/kibana/commit/6a71bd959d864b132bf05f4ed33d921dc8163462) | diff |
| --- | --- | --- | --- |
| `core` | 444.6KB | 444.4KB | -219.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/45f3241db0bd06a264a94d8cfd104bf4057db77e) | [after](https://github.com/elastic/kibana/commit/6a71bd959d864b132bf05f4ed33d921dc8163462) | diff |
| --- | --- | --- | --- |
| `@kbn/core-chrome-browser` | 211 | 210 | -1 |


</details>


### History
* :yellow_heart: [Build #265197](https://buildkite.com/elastic/kibana-pull-request/builds/265197) was flaky 964c1ca0fdb54c13ab25a6c80c1ad86ea774fbfb
* :broken_heart: [Build #265096](https://buildkite.com/elastic/kibana-pull-request/builds/265096) failed de26711c23519e22f42ad687e5d66ffd48f92b0d
* :broken_heart: [Build #264884](https://buildkite.com/elastic/kibana-pull-request/builds/264884) failed 8252651a2421bbf421b631a79776c753cdeeaf7b

cc @paulinashakirova

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265649","number":265649,"commit":"6a71bd959d864b132bf05f4ed33d921dc8163462"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265197","number":265197,"commit":"964c1ca0fdb54c13ab25a6c80c1ad86ea774fbfb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265096","number":265096,"commit":"de26711c23519e22f42ad687e5d66ffd48f92b0d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264884","number":264884,"commit":"8252651a2421bbf421b631a79776c753cdeeaf7b"}],"number":265649}
buildkite-pr-comment-->

Commit:
revert to use spacer because of found limitation

Commit:
revert to use spacer because of found limitation

Commit:
using EuiTheme from context with class comp

Commit:
using EuiTheme from context with class comp

Commit:
Merge branch 'main' into consistent-help-dropdown-ui-2

paulinashakirova commented:
> @ek-so Yes, we can remove those spacers ☺️ I pushed the commit about it now. ✅

Writing here for visibility, that In the end, turns out that removal of those Spacers introduces other issues, fixing which (with EuiSpacers) would involve more customization. So we agreed to keep on using EuiSpacer between buttons.

paulinashakirova commented on src/core/packages/chrome/browser-internal/src/ui/header/header_help_menu.tsx:
```diff
@@ -92,10 +93,10 @@ interface State {
   private subscription?: Subscription;
 
-  constructor(props: Props) {
+  constructor(props: Props & WithEuiThemeProps) {
```
Had a thought about extracting this union in a separate type and reuse it, but it is only repeated twice, so maybe it is not worth it. WDYT?

Dosant replied:
I think this is fine like this

jasonrhodes committed:
Adds notification settings for unsupported ftr tests (synthetics, inventory) (#206085)

## Summary

Based on [work by @v1v to notify Slack teams when there are unsupported
FTR failures](https://github.com/elastic/kibana/pull/205260), I've added
the remaining ones for synthetics and inventory tests here.

markov00 committed:
[Lens] Align Lens style with Borealis (#204839)

## Summary

This PR updates the style of Lens to align it with the new Borealis
theme.
It covers the set of tasks in #203050 within the `Lens editor panels`
list.

Comments are applied in order as in the mentioned issue

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

legrego committed:
Dependency ownership refinements (#205937)

## Summary

This PR assigns owners for dependencies that are not yet claimed. These
assignments were a "best effort", and will likely need tweaking once
downstream teams start receiving renovate PRs.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/265756)
* Commit: 33b96af5880425185d589b62e6f0948efe655979

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/b25c9984bb4ae9e7bb2b17641b699c8c8844f3de) | [after](https://github.com/elastic/kibana/commit/33b96af5880425185d589b62e6f0948efe655979) | diff |
| --- | --- | --- | --- |
| `@kbn/core-chrome-browser` | 104 | 103 | -1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/b25c9984bb4ae9e7bb2b17641b699c8c8844f3de) | [after](https://github.com/elastic/kibana/commit/33b96af5880425185d589b62e6f0948efe655979) | diff |
| --- | --- | --- | --- |
| `core` | 444.6KB | 444.6KB | -39.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/b25c9984bb4ae9e7bb2b17641b699c8c8844f3de) | [after](https://github.com/elastic/kibana/commit/33b96af5880425185d589b62e6f0948efe655979) | diff |
| --- | --- | --- | --- |
| `@kbn/core-chrome-browser` | 211 | 210 | -1 |


</details>


### History
* :yellow_heart: [Build #265649](https://buildkite.com/elastic/kibana-pull-request/builds/265649) was flaky 6a71bd959d864b132bf05f4ed33d921dc8163462
* :yellow_heart: [Build #265197](https://buildkite.com/elastic/kibana-pull-request/builds/265197) was flaky 964c1ca0fdb54c13ab25a6c80c1ad86ea774fbfb
* :broken_heart: [Build #265096](https://buildkite.com/elastic/kibana-pull-request/builds/265096) failed de26711c23519e22f42ad687e5d66ffd48f92b0d
* :broken_heart: [Build #264884](https://buildkite.com/elastic/kibana-pull-request/builds/264884) failed 8252651a2421bbf421b631a79776c753cdeeaf7b

cc @paulinashakirova

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265756","number":265756,"commit":"33b96af5880425185d589b62e6f0948efe655979"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265649","number":265649,"commit":"6a71bd959d864b132bf05f4ed33d921dc8163462"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265197","number":265197,"commit":"964c1ca0fdb54c13ab25a6c80c1ad86ea774fbfb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265096","number":265096,"commit":"de26711c23519e22f42ad687e5d66ffd48f92b0d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264884","number":264884,"commit":"8252651a2421bbf421b631a79776c753cdeeaf7b"}],"number":265756}
buildkite-pr-comment-->

bhapas committed:
[Automatic Import ] Enable inference connector for Auto Import (#206111)

## Summary

Enables new inference connector in the Automatic Import.

This PR also fixes the use of `inferenceEnabled` from
`useAssistantContext` since it is not available in AutoImport.

## To test

1. Update the value for `inferenceConnectorOn` to `true` in
`x-pack/platform/plugins/shared/stack_connectors/common/experimental_features.ts`
2. Create an inference connector using [OpenAI
creds](https://p.elstc.co/paste/36VivuC+#TnP7-Z7wBKDUg8fQ/lTycSCdwUxEEbHcyQ/Q0i3oEmO).
Configure the inference endpoint for completion and name the endpoint
`openai-completion-preconfig`
3. Now that the inference endpoint is created, add a [preconfigured
connector](https://p.elstc.co/paste/tFWF3LSA#0thBRW05e6KSSkLCDjQiH8GkECQySBiHm6zRMCUThlf)
with the same credentials.
4. Select the preconfigured selector in Automatic Import.
5. Test the Auto Import flow works.

---------

Co-authored-by: Steph Milovic <stephanie.milovic@elastic.co>

delanni committed:
[ci] Work around docosaurus errors (#206097)

## Summary
This workaround removes the folder that needs to be built differently
after cloning the repos.

See: https://github.com/elastic/kibana/issues/206077

nickofthyme committed:
[User] Prevent disabled input change by password manager (#204269)

Prevents username input from being edited by password manager extensions when `disabled`

davismcphee committed:
[Discover] Format JSON messages in Observability Logs profile (#205666)

## Summary

This PR updates the Observability Logs profile to detect and auto format
JSON message values within both the Log overview doc viewer tab and the
Summary cell popover. Additionally, it enables CTRL/CMD + F find
functionality within the doc viewer JSON tab for all contexts to make it
easier for users to search the JSON output.

JSON message formatting:

![json](https://github.com/user-attachments/assets/a7c63afd-bef7-4050-b8cf-08e4f469ffa9)

JSON tab find functionality:

![find](https://github.com/user-attachments/assets/aac51e05-6126-4770-8976-0d9057bad557)

### Checklist

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

kibanamachine committed:
[api-docs] 2025-01-10 Daily api_docs build (#206196)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/948

stratoula committed:
[ES|QL] Allows the editor suggestions to be visible when inline docs flyout is open (#206064)

## Summary

Closes https://github.com/elastic/kibana/issues/199299

Changes the flyout z-index in order the suggestions to be visible when
the docs are open.


![image
(75)](https://github.com/user-attachments/assets/e13595d0-e98c-4e7b-9673-a6e3d011fee7)

afharo committed:
[Cloud Experiments] Flag as private (#206121)

mykolaharmash committed:
Migrate onboarding Cypress tests to Scout (#205482)

This change converts Cypress tests for the custom logs flow into
Playwright using [the Scout
wrapper](https://github.com/elastic/kibana/tree/main/packages/kbn-scout).

> [!NOTE]
> As Scout package is still being developed, the PR pipeline configured
to runs Playwright tests only when code in certain plugins have been
changed and not on every PR.

### How to run tests locally

Start the Scout server
```bash
node scripts/scout.js start-server --stateful
```

In a separate terminal run the tests
```bash
npx playwright test --config x-pack/solutions/observability/plugins/observability_onboarding/ui_tests/playwright.config.ts
```

Playwright runs browsers in a headless mode by default, user `--headed`
option if needed

dej611 committed:
[Lens] Restore embeddable test coverage (#204986)

## Summary

Fixes #198754

Restore previous removed tests when performing the refactor.
The new tests take advantage of the new architecture to be more modular
and close to the logic modules.

The `data_loader` tests are not just covering the re-render logic but
also some `expression_params` logic, who in the past have proven to be
the source of some bugs: specifically the tests will check that the
params are correctly passed to the params logic and then stored
correctly in the observable.

New mocks take advantage of the plain initializers to build some of the
API, that will make it in sync with the actual implementation for future
maintenance.


### Checklist

Check the PR satisfies following conditions. 

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Marta Bondyra <4283304+mbondyra@users.noreply.github.com>

Dosant reviewed (approved):
lgtm!

dej611 committed:
[Lens][Security Solutions] Restore cypress tests (#206025)

## Summary

Fixes #198756 


### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

juliaElastic committed:
[Fleet] fix bulk actions timing out sometimes (#205735)

## Summary

Closes https://github.com/elastic/ingest-dev/issues/4346

Update: changed the implementation to run the first attempt of bulk
action execution in the task too.

```
[2025-01-08T11:10:54.139+01:00][INFO ][plugins.fleet] Running action asynchronously, actionId: f8658178-cb1e-485d-9d2f-ad60ccc37bc9, total agents:10001
actionParams {
  kuery: '( fleet-agents.policy_id : ("d3448ae1-9e55-485e-b74c-83471cb06977")) and (status:online or (status:error or status:degraded) or (status:updating or status:unenrolling or status:enrolling) or status:offline)',
  revoke: false,
  force: undefined,
  batchSize: 10000,
  showInactive: true,
  spaceId: 'default',
  total: 10001,
  actionId: 'f8658178-cb1e-485d-9d2f-ad60ccc37bc9'
}
retryParams {
  pitId: 'gIuaBAEPLmZsZWV0LWFnZW50cy03FmRQN3pDT1gzUnFpNFkwdHJFdzJvbncAARZ5N2R5SnVSelN2bWNMang1THNfVkNRAAEAAAAAAAFpZhZiZnpybjZYSFRRNklYNlBqWHFzVU1nAAEWZFA3ekNPWDNScWk0WTB0ckV3Mm9udwAA',
  retryCount: 1
}
[2025-01-08T11:10:54.154+01:00][INFO ][plugins.fleet] Scheduling task fleet:unenroll_action:retry:f8658178-cb1e-485d-9d2f-ad60ccc37bc9
[2025-01-08T11:10:54.154+01:00][DEBUG][plugins.fleet] [Bulk Agent Unenroll API] Unenroll agents in 647ms
[2025-01-08T11:10:55.772+01:00][WARN ][http.server.Kibana] Event loop utilization for /julia/api/fleet/agents exceeded threshold of 250ms (351ms out of 633ms) and 15% (55%) 
[2025-01-08T11:10:57.235+01:00][INFO ][plugins.fleet] Running bulk action retry task
[2025-01-08T11:10:57.235+01:00][DEBUG][plugins.fleet] Running task fleet:unenroll_action:retry:f8658178-cb1e-485d-9d2f-ad60ccc37bc9
[2025-01-08T11:10:57.236+01:00][INFO ][plugins.fleet] Completed bulk action retry task
[2025-01-08T11:10:57.251+01:00][INFO ][plugins.fleet] Scheduling task fleet:unenroll_action:retry:check:f8658178-cb1e-485d-9d2f-ad60ccc37bc9
[2025-01-08T11:10:57.251+01:00][DEBUG][plugins.fleet] kuery: ( fleet-agents.policy_id : ("d3448ae1-9e55-485e-b74c-83471cb06977")) and (status:online or (status:error or status:degraded) or (status:updating or status:unenrolling or status:enrolling) or status:offline)
[2025-01-08T11:10:57.781+01:00][DEBUG][plugins.fleet] Action not found
[2025-01-08T11:10:58.891+01:00][DEBUG][plugins.fleet] Secrets storage requirements already met, turned on in settings
[2025-01-08T11:10:59.414+01:00][WARN ][http.server.Kibana] Event loop utilization for /julia/api/fleet/agent_status exceeded threshold of 250ms (294ms out of 348ms) and 15% (85%) 
[2025-01-08T11:10:59.806+01:00][WARN ][http.server.Kibana] Event loop utilization for /julia/api/fleet/agents exceeded threshold of 250ms (504ms out of 743ms) and 15% (68%) 
[2025-01-08T11:11:01.532+01:00][INFO ][plugins.fleet] Removing task fleet:unenroll_action:retry:check:f8658178-cb1e-485d-9d2f-ad60ccc37bc9
actionParams {
  kuery: '( fleet-agents.policy_id : ("d3448ae1-9e55-485e-b74c-83471cb06977")) and (status:online or (status:error or status:degraded) or (status:updating or status:unenrolling or status:enrolling) or status:offline)',
  revoke: false,
  batchSize: 10000,
  showInactive: true,
  spaceId: 'default',
  total: 10001,
  actionId: 'f8658178-cb1e-485d-9d2f-ad60ccc37bc9'
}
retryParams {
  pitId: 'gIuaBAEPLmZsZWV0LWFnZW50cy03FmRQN3pDT1gzUnFpNFkwdHJFdzJvbncAARZ5N2R5SnVSelN2bWNMang1THNfVkNRAAEAAAAAAAFpZhZiZnpybjZYSFRRNklYNlBqWHFzVU1nAAEWZFA3ekNPWDNScWk0WTB0ckV3Mm9udwAA',
  retryCount: 1,
  taskId: 'fleet:unenroll_action:retry:f8658178-cb1e-485d-9d2f-ad60ccc37bc9',
  searchAfter: [ 1736331016589, 'online-98', 119770 ]
}
[2025-01-08T11:11:01.564+01:00][INFO ][plugins.fleet] Scheduling task fleet:unenroll_action:retry:check:f8658178-cb1e-485d-9d2f-ad60ccc37bc9
[2025
```


## Old description

Bulk actions supposed to run async in a kibana task, and the API to
return quickly with an action id.
This was implemented
[here](https://github.com/elastic/kibana/commit/38e74d7ee6f1089c7c4857b3ad320849ea782415#diff-69fa063ab8857204486203a718ff4bd0cbf9652623279d1959316de8e83233ff)
and unintentionally broke when a tslint rule was introduced
[here](https://github.com/elastic/kibana/pull/181456/files#diff-69fa063ab8857204486203a718ff4bd0cbf9652623279d1959316de8e83233ff),
effectively letting the async code finish before the API returns. This
results in the API timing out sometimes when there are many agents.

Tested by creating 100k agent docs with the `create_agents` script and
bulk unenroll agents.

Logs before the change:
```
[2025-01-07T14:38:04.467+01:00][INFO ][plugins.fleet] Running action asynchronously, actionId: 2a57ac7a-0c1b-4a08-8709-6ccb683db995, total agents:100000
[2025-01-07T14:38:04.482+01:00][INFO ][plugins.fleet] Scheduling task fleet:unenroll_action:retry:check:2a57ac7a-0c1b-4a08-8709-6ccb683db995
[2025-01-07T14:38:04.482+01:00][DEBUG][plugins.fleet] kuery: status:* AND (fleet-agents.policy_id : ("d3448ae1-9e55-485e-b74c-83471cb06977"))
...
[2025-01-07T14:39:00.264+01:00][INFO ][plugins.fleet] Removing task fleet:unenroll_action:retry:check:2a57ac7a-0c1b-4a08-8709-6ccb683db995
[2025-01-07T14:39:00.290+01:00][INFO ][plugins.fleet] Scheduling task fleet:unenroll_action:retry:check:2a57ac7a-0c1b-4a08-8709-6ccb683db995
[2025-01-07T14:39:00.293+01:00][INFO ][plugins.fleet] processed 100000 agents, took 55811ms
[2025-01-07T14:39:00.293+01:00][INFO ][plugins.fleet] Removing task fleet:unenroll_action:retry:check:2a57ac7a-0c1b-4a08-8709-6ccb683db995
[2025-01-07T14:39:00.304+01:00][DEBUG][plugins.fleet] [Bulk Agent Unenroll API] Unenroll agents in 56027ms
```

Logs after the change:
```
[2025-01-07T14:42:10.178+01:00][INFO ][plugins.fleet] Running action asynchronously, actionId: c3e12928-bbfe-4731-bdc2-c47bcac727a7, total agents:100000
[2025-01-07T14:42:10.194+01:00][INFO ][plugins.fleet] Scheduling task fleet:unenroll_action:retry:check:c3e12928-bbfe-4731-bdc2-c47bcac727a7
[2025-01-07T14:42:10.195+01:00][DEBUG][plugins.fleet] kuery: status:* AND (fleet-agents.policy_id : ("d3448ae1-9e55-485e-b74c-83471cb06977"))
[2025-01-07T14:42:10.195+01:00][DEBUG][plugins.fleet] [Bulk Agent Unenroll API] Unenroll agents in 196ms
[2025-01-07T14:43:00.762+01:00][INFO ][plugins.fleet] processed 100000 agents, took 50567ms
[2025-01-07T14:43:00.762+01:00][INFO ][plugins.fleet] Removing task fleet:unenroll_action:retry:check:c3e12928-bbfe-4731-bdc2-c47bcac727a7
```

tomsonpl committed:
[EDR Workflows] Add Runscript openApi schema  (#206044)

js-jankisalvi committed:
[ResponseOps][Cases] Make case templates GA (#205940)

## Summary

This PR makes case templates GA.

<details><summary>Case settings</summary>

<img width="1133" alt="image"
src="https://github.com/user-attachments/assets/45c5896f-88dc-4199-ab83-398d05c8746a"
/>
</details> 

<details><summary>Case create</summary>

<img width="1094" alt="image"
src="https://github.com/user-attachments/assets/05df0a4a-44e8-4bf2-ad8d-e50f5d9851ef"
/>
</details> 

<details><summary>Case action</summary>

<img width="1079" alt="image"
src="https://github.com/user-attachments/assets/b321e926-3b2d-4773-90a3-3bd0b45aac3c"
/>
</details> 

### Checklist

Check the PR satisfies following conditions. 

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

### Release Notes
Case templates are now GA

### For maintainers

- [x] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

maryam-saeidi committed:
Remove wrong test (#206231)

## Summary

This test is not useful and has a bug, so we decided to remove it.

gsoldevila committed:
SKA: Relocate Script v7.1 (#206233)

## Summary

* Fix an issue with the `--list` command failing the 1st run.
* Allow passing in no filters, and relocate "incorrect" modules (aka
modules that are not in the correct folder) in that case.

bryce-b committed:
[INFRA] Add additional perf telemetry  (#205955)

## Summary
Closes https://github.com/elastic/kibana/issues/205394
This PR adds performance telemetry to metric explorer and adds time
range to existing telemetry in the host tables.
Building on #180309.

The time ranges added to the telemetry use the format "now", "now-15m"
per the examples in the [performance tutorial
docs](https://docs.elastic.dev/kibana-dev-docs/tutorial/performance/adding_custom_performance_metrics#add-time-ranges)


example of perf temeletry with the query range metadata can be seen
[here](https://telemetry-v2-staging.elastic.dev/s/apm/app/discover#/?_g=(filters:!(),query:(language:kuery,query:''),refreshInterval:(pause:!t,value:60000),time:(from:'2025-01-08T18:30:00.000Z',to:'2025-01-08T19:00:00.000Z'))&_a=(columns:!(),dataSource:(dataViewId:'0d6d7d31-1369-4a53-b36d-fbe97e4e5a0e',type:dataView),filters:!(),interval:auto,query:(language:kuery,query:'eventName%20:%20%22kibana:plugin_render_time%22%20and%20context.pageName%20:%20%22application:metrics:%2Fexplorer%22%20'),sort:!(!(timestamp,desc))))

tonyghiani committed:
[Streams 🌊] Stream enrichment processors management (#204793)

## 📓 Summary

Part of #https://github.com/elastic/streams-program/issues/32

This work implements a UI for basic stream enrichment, supporting grok
and dissect processor + detected fields mapping.

The main features implemented in this PR consist of:
- **Sortable processors list**
- **Add new processor - Grok, Dissect**
  - Ad-hoc forms for each processor
  - Simulated document outcome with extracted fields
  - Filter matching documents with parsed fields
  - Mapping detected fields (only available for wired streams)
- **Edit processor**
  - Change configuration only
  - Delete processor CTA

As a side quest, I added a small package for object utils as
@simianhacker suggested.
`@kbn/object-utils` exposes `calculateObjectDiff` and `flattenObject` to
detect the changed fields in a simulation.

## 🔜 Follow-up work

I'll work on minor updates on top of this MVP to make this available for
further testing from the team.
The next steps will be:
- **Tests** for features that consolidate on the functional pov.
- Better field mapping detection and UI feedback (infer the type of the
detected field, currently always unmapped)
- Add better form validation and feedback for processor configuration.

As discussed offline, state management is purely based on the built-in
react APIs + react-hook-form. It could be improved with different
approaches, including a more solid state management library to make it
easier to maintain and bulletproof to race conditions. No state syncs
with the URL currently.

## 🎥 Demo


https://github.com/user-attachments/assets/a48fade9-f5aa-4270-bb19-d91d1eed822b

---------

Co-authored-by: Marco Antonio Ghiani <marcoantonio.ghiani@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

artem-shelkovnikov committed:
Introduce Kibana task to deploy agentless connectors for 9.0 (#203973)

## Closes https://github.com/elastic/search-team/issues/8508
## Closes https://github.com/elastic/search-team/issues/8465

## Summary

This PR adds a background task for search_connectors plugin. This task
checks connector records and agentless package policies and sees if new
connector was added/old was deleted, and then adds/deletes package
policies for these connectors.

Scenario 1: a new connector was added by a user/API call

User creates an Elastic-managed connector:


https://github.com/user-attachments/assets/38296e48-b281-4b2b-9750-ab0a47334b55

When the user is done, a package policy is created by this background
task:


https://github.com/user-attachments/assets/12dbc33f-32bf-472d-b854-64588fc1e5b1

Scenario 2: a connector was deleted by a user/API call

User deletes an Elastic-managed connector:


https://github.com/user-attachments/assets/5997897e-fb9d-4199-8045-abe163264976

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Jedr Blaszyk <jedrazb@gmail.com>

iblancof committed:
[APM] Fix Cypress flaky test in Custom Links (#206108)

## Summary

Fixes https://github.com/elastic/kibana/issues/206016

This PR aims to fix a flaky test that waits for an empty state screen,
which sometimes fails to appear.

The issue might happen because the data deletion action, which triggers
the empty state, is not completed properly. This action takes place in
the previous test and does not wait for the deletion to finish.

The proposed solution ensures that the test responsible for deleting the
data waits until the empty state appears after deletion, preventing the
next test from running too soon.

### Test locally

`````
node x-pack/solutions/observability/plugins/apm/scripts/test/e2e.js --server
node x-pack/solutions/observability/plugins/apm/scripts/test/e2e.js  --runner --open
`````

Run the
[custom_links.cy.ts](http://localhost:5620/__/#/specs/runner?file=cypress/e2e/settings/custom_links.cy.ts)
test.

Dosant committed:
Sustainable Kibana Architecture: Move the rest of shared-ux modules (#205924)

## Summary

This PR aims at relocating some of the Kibana modules (plugins and
packages) into a new folder structure, according to the _Sustainable
Kibana Architecture_ initiative.

> [!IMPORTANT]
> * We kindly ask you to:
> * Manually fix the errors in the error section below (if there are
any).
> * Search for the `packages[\/\\]` and `plugins[\/\\]` patterns in the
source code (Babel and Eslint config files), and update them
appropriately.
> * Manually review
`.buildkite/scripts/pipelines/pull_request/pipeline.ts` to ensure that
any CI pipeline customizations continue to be correctly applied after
the changed path names
> * Review all of the updated files, specially the `.ts` and `.js` files
listed in the sections below, as some of them contain relative paths
that have been updated.
> * Think of potential impact of the move, including tooling and
configuration files that can be pointing to the relocated modules. E.g.:
>     * customised eslint rules
>     * docs pointing to source code

> [!NOTE]
> * This PR has been auto-generated.
> * Any manual contributions will be lost if the 'relocate' script is
re-run.
> * Try to obtain the missing reviews / approvals before applying manual
fixes, and/or keep your changes in a .patch / git stash.
> * Please use
[#sustainable_kibana_architecture](https://elastic.slack.com/archives/C07TCKTA22E)
Slack channel for feedback.

Are you trying to rebase this PR to solve merge conflicts? Please follow
the steps describe
[here](https://elastic.slack.com/archives/C07TCKTA22E/p1734019532879269?thread_ts=1734019339.935419&cid=C07TCKTA22E).

#### 1 plugin(s) are going to be relocated:

| Id | Target folder |
| -- | ------------- |
| `@kbn/url-drilldown-plugin` |
`x-pack/platform/plugins/private/drilldowns/url_drilldown` |




#### 21 packages(s) are going to be relocated:

| Id | Target folder |
| -- | ------------- |
| `@kbn/core-chrome-browser` | `src/core/packages/chrome/browser` |
| `@kbn/deeplinks-shared` |
`src/platform/packages/shared/deeplinks/shared` |
| `@kbn/home-sample-data-card` |
`src/platform/packages/shared/home/sample_data_card` |
| `@kbn/home-sample-data-tab` |
`src/platform/packages/shared/home/sample_data_tab` |
| `@kbn/home-sample-data-types` |
`src/platform/packages/shared/home/sample_data_types` |
| `@kbn/guided-onboarding` |
`src/platform/packages/shared/kbn-guided-onboarding` |
| `@kbn/item-buffer` | `src/platform/packages/private/kbn-item-buffer` |
| `@kbn/management-settings-section-registry` |
`src/platform/packages/shared/kbn-management/settings/section_registry`
|
| `@kbn/management-settings-ids` |
`src/platform/packages/shared/kbn-management/settings/setting_ids` |
| `@kbn/react-mute-legacy-root-warning` |
`src/platform/packages/private/kbn-react-mute-legacy-root-warning` |
| `@kbn/saved-objects-settings` |
`src/platform/packages/private/kbn-saved-objects-settings` |
| `@kbn/react-kibana-context-common` |
`src/platform/packages/shared/react/kibana_context/common` |
| `@kbn/react-kibana-context-render` |
`src/platform/packages/shared/react/kibana_context/render` |
| `@kbn/react-kibana-context-root` |
`src/platform/packages/shared/react/kibana_context/root` |
| `@kbn/react-kibana-context-styled` |
`src/platform/packages/shared/react/kibana_context/styled` |
| `@kbn/react-kibana-context-theme` |
`src/platform/packages/shared/react/kibana_context/theme` |
| `@kbn/react-kibana-mount` |
`src/platform/packages/shared/react/kibana_mount` |
| `@kbn/serverless-project-switcher` |
`src/platform/packages/private/serverless/project_switcher` |
| `@kbn/serverless-common-settings` |
`src/platform/packages/private/serverless/settings/common` |
| `@kbn/serverless-observability-settings` |
`src/platform/packages/shared/serverless/settings/observability_project`
|
| `@kbn/serverless-types` |
`src/platform/packages/private/serverless/types` |


<details >
<summary>Updated relative paths</summary>

```
src/core/packages/chrome/browser/jest.config.js:12
src/core/packages/chrome/browser/tsconfig.json:2
src/core/packages/chrome/browser/tsconfig.type_check.json:2
src/core/packages/chrome/browser/tsconfig.type_check.json:21
src/core/packages/chrome/browser/tsconfig.type_check.json:24
src/core/packages/chrome/browser/tsconfig.type_check.json:27
src/core/packages/chrome/browser/tsconfig.type_check.json:30
src/core/packages/chrome/browser/tsconfig.type_check.json:33
src/core/packages/chrome/browser/tsconfig.type_check.json:36
src/core/packages/chrome/browser/tsconfig.type_check.json:39
src/core/packages/chrome/browser/tsconfig.type_check.json:42
src/core/packages/chrome/browser/tsconfig.type_check.json:45
src/core/packages/chrome/browser/tsconfig.type_check.json:48
src/core/packages/chrome/browser/tsconfig.type_check.json:51
src/core/packages/chrome/browser/tsconfig.type_check.json:54
src/platform/packages/private/kbn-item-buffer/jest.config.js:12
src/platform/packages/private/kbn-item-buffer/tsconfig.json:2
src/platform/packages/private/kbn-react-mute-legacy-root-warning/jest.config.js:12
src/platform/packages/private/kbn-react-mute-legacy-root-warning/tsconfig.json:2
src/platform/packages/private/kbn-saved-objects-settings/jest.config.js:12
src/platform/packages/private/kbn-saved-objects-settings/tsconfig.json:2
src/platform/packages/private/kbn-saved-objects-settings/tsconfig.type_check.json:2
src/platform/packages/private/serverless/project_switcher/jest.config.js:12
src/platform/packages/private/serverless/project_switcher/tsconfig.json:2
src/platform/packages/private/serverless/project_switcher/tsconfig.type_check.json:2
src/platform/packages/private/serverless/project_switcher/tsconfig.type_check.json:23
src/platform/packages/private/serverless/settings/common/tsconfig.json:2
src/platform/packages/private/serverless/settings/common/tsconfig.type_check.json:2
src/platform/packages/private/serverless/settings/common/tsconfig.type_check.json:20
src/platform/packages/private/serverless/types/tsconfig.json:2
src/platform/packages/private/serverless/types/tsconfig.type_check.json:2
src/platform/packages/shared/deeplinks/shared/jest.config.js:12
src/platform/packages/shared/deeplinks/shared/tsconfig.json:2
src/platform/packages/shared/deeplinks/shared/tsconfig.type_check.json:2
src/platform/packages/shared/home/sample_data_card/jest.config.js:12
src/platform/packages/shared/home/sample_data_card/tsconfig.json:2
src/platform/packages/shared/home/sample_data_card/tsconfig.type_check.json:2
src/platform/packages/shared/home/sample_data_card/tsconfig.type_check.json:23
src/platform/packages/shared/home/sample_data_card/tsconfig.type_check.json:29
src/platform/packages/shared/home/sample_data_tab/jest.config.js:12
src/platform/packages/shared/home/sample_data_tab/tsconfig.json:2
src/platform/packages/shared/home/sample_data_tab/tsconfig.type_check.json:2
src/platform/packages/shared/home/sample_data_tab/tsconfig.type_check.json:23
src/platform/packages/shared/home/sample_data_types/jest.config.js:12
src/platform/packages/shared/home/sample_data_types/tsconfig.json:2
src/platform/packages/shared/home/sample_data_types/tsconfig.type_check.json:2
src/platform/packages/shared/kbn-guided-onboarding/jest.config.js:12
src/platform/packages/shared/kbn-guided-onboarding/tsconfig.json:2
src/platform/packages/shared/kbn-guided-onboarding/tsconfig.type_check.json:2
src/platform/packages/shared/kbn-management/settings/section_registry/tsconfig.json:2
src/platform/packages/shared/kbn-management/settings/section_registry/tsconfig.type_check.json:2
src/platform/packages/shared/kbn-management/settings/section_registry/tsconfig.type_check.json:22
src/platform/packages/shared/kbn-management/settings/section_registry/tsconfig.type_check.json:25
src/platform/packages/shared/kbn-management/settings/setting_ids/tsconfig.json:2
src/platform/packages/shared/kbn-management/settings/setting_ids/tsconfig.type_check.json:2
src/platform/packages/shared/react/kibana_context/common/jest.config.js:12
src/platform/packages/shared/react/kibana_context/common/tsconfig.json:2
src/platform/packages/shared/react/kibana_context/common/tsconfig.type_check.json:2
src/platform/packages/shared/react/kibana_context/render/jest.config.js:12
src/platform/packages/shared/react/kibana_context/render/tsconfig.json:2
src/platform/packages/shared/react/kibana_context/render/tsconfig.type_check.json:2
src/platform/packages/shared/react/kibana_context/render/tsconfig.type_check.json:25
src/platform/packages/shared/react/kibana_context/root/jest.config.js:12
src/platform/packages/shared/react/kibana_context/root/tsconfig.json:2
src/platform/packages/shared/react/kibana_context/root/tsconfig.type_check.json:2
src/platform/packages/shared/react/kibana_context/root/tsconfig.type_check.json:22
src/platform/packages/shared/react/kibana_context/root/tsconfig.type_check.json:28
src/platform/packages/shared/react/kibana_context/root/tsconfig.type_check.json:31
src/platform/packages/shared/react/kibana_context/root/tsconfig.type_check.json:34
src/platform/packages/shared/react/kibana_context/root/tsconfig.type_check.json:37
src/platform/packages/shared/react/kibana_context/root/tsconfig.type_check.json:40
src/platform/packages/shared/react/kibana_context/styled/jest.config.js:12
src/platform/packages/shared/react/kibana_context/styled/tsconfig.json:2
src/platform/packages/shared/react/kibana_context/styled/tsconfig.type_check.json:2
src/platform/packages/shared/react/kibana_context/styled/tsconfig.type_check.json:22
src/platform/packages/shared/react/kibana_context/theme/jest.config.js:12
src/platform/packages/shared/react/kibana_context/theme/tsconfig.json:2
src/platform/packages/shared/react/kibana_context/theme/tsconfig.type_check.json:2
src/platform/packages/shared/react/kibana_context/theme/tsconfig.type_check.json:22
src/platform/packages/shared/react/kibana_mount/jest.config.js:12
src/platform/packages/shared/react/kibana_mount/tsconfig.json:2
src/platform/packages/shared/react/kibana_mount/tsconfig.type_check.json:2
src/platform/packages/shared/react/kibana_mount/tsconfig.type_check.json:22
src/platform/packages/shared/react/kibana_mount/tsconfig.type_check.json:25
src/platform/packages/shared/react/kibana_mount/tsconfig.type_check.json:28
src/platform/packages/shared/react/kibana_mount/tsconfig.type_check.json:34
src/platform/packages/shared/serverless/settings/observability_project/tsconfig.json:2
src/platform/packages/shared/serverless/settings/observability_project/tsconfig.type_check.json:2
src/platform/packages/shared/serverless/settings/observability_project/tsconfig.type_check.json:20
x-pack/platform/plugins/private/drilldowns/url_drilldown/tsconfig.json:2
```

</details>

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

gbamparop committed:
Remove log stream and settings (#204115)

Removes the code used to render the log stream and settings pages.

The following areas have been checked:
- Log stream embeddable (dashboard of the `cisco_meraki` integration)
- Log stream shared component
- Log categories
- Log anomalies


https://github.com/user-attachments/assets/2bc0763d-3def-4c4b-b50a-21c17976a596

Closes https://github.com/elastic/kibana/issues/204005

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

artem-shelkovnikov committed:
Add new native fields for SPO connector - certificate authenticate (#205337)

## Summary

This PR reflects the changes done in
https://github.com/elastic/connectors/pull/3064: update of Sharepoint
Online native connector configurable fields.

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

Commit:
Merge branch 'consistent-help-dropdown-ui-2' of https://github.com/paulinashakirova/kibana into consistent-help-dropdown-ui-2
